### PR TITLE
CA1816: correct idisposable implementation

### PR DESCRIPTION
**Pull request by Evangelink**
Fix implementation of the disposable pattern.

## Description
See [Implementing Dispose](https://docs.microsoft.com//dotnet/standard/garbage-collection/implementing-dispose) and [CA1816](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1816)

### CODE REVIEW

File: src/GitTools.Testing/Fixtures/RepositoryFixtureBase.cs [hunk 1]
```diff
@@ -33,8 +33,19 @@ protected RepositoryFixtureBase(IRepository repository)
         /// <summary>
         ///     Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
         /// </summary>
-        public virtual void Dispose()
+        public void Dispose()
```
Comment (Evangelink) [inline src/GitTools.Testing/Fixtures/RepositoryFixtureBase.cs:36]: "Proper" implementation of the disposable pattern.

File: src/GitVersion.Core/VersionConverters/AssemblyInfo/AssemblyInfoFileUpdater.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@ public interface IAssemblyInfoFileUpdater : IVersionConverter<AssemblyInfoContex
     {
     }
 
-    public class AssemblyInfoFileUpdater : IAssemblyInfoFileUpdater
+    public sealed class AssemblyInfoFileUpdater : IAssemblyInfoFileUpdater
```
Comment (Evangelink) [inline src/GitVersion.Core/VersionConverters/AssemblyInfo/AssemblyInfoFileUpdater.cs:16]: When class wasn't inherited I went with the option of sealing the class but we could also go for the disposable pattern implementation.

Note that this change could be considered as breaking change if you consider this was part of your API.

**Review by asbjornu — APPROVED** [review]
I think this is fine. I can't imagine anyone using `GitTools.Testing` beside us, so I don't think there are any consumers to care for. If there are, perhaps with this change we'll find out.

**Pull request merged**